### ISSUE TITLE
Restart lopp, uncaught exception

### ISSUE THREAD

**Issue by tipp88** [unltdnetworx/ioBroker.stiebel-isg#63]
Is this Adapter still maintained?
Adapter frequently crashes due to unhandled exception and stucks in restart loop.
![image](https://user-images.githubusercontent.com/68737433/201330338-fae74853-8544-4d02-9028-83aa9f37a7be.png)

**Comment by unltdnetworx** [comment]
It is maintained, but I'm a little busy till march 2023.

I think this problem was already a fixed thread in the iobroker forum.

**Comment by unltdnetworx** [comment]
@tipp88 
Can you reach the web-interface with your entered ip-adress?

**Comment by tipp88** [comment]
> @tipp88 Can you reach the web-interface with your entered ip-adress?

Yes, communication to the Web interface in general works and the adapter reads the data.
The error seems to disappear if i deactivate "enable expert settings" in the adapter config

**Comment by chmaod** [comment]
Have the same problem here
Version 1.7.3 in /opt/iobroker/node_modules/iobroker.stiebel-isg, node: v14.20.1, js-controller: 4.0.23

Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().
unhandled promise rejection: read ECONNRESET

In the down-time-window I tried to get answer from the webinterface, this is normally working, and also checked all network logs, no issues. It must have to do with the instance. 
Any idea to fix?

**Comment by unltdnetworx** [comment]
Ist das Problem seit dem letzten Update aufgetreten?

**Comment by unltdnetworx** [comment]
Kann es sein, dass das ISG ein Update bekommen hat? Lief der Adapter bisher?
@tipp88 Evtl. mal in den URL-Einstellungen des Adapters die abgefragten Seiten der "Experten-Einstellungen" entfernen (vorher kopieren), sehen ob der Adapter dann normal läuft und dann nach und nach wieder einfügen. Mal beobachten, ob es an bestimmten Seiten liegt und diese dann einfach nicht mehr abfragen/weglassen.

**Comment by chmaod** [comment]
Nein, das ISG hat garantiert kein Update erhalten, da ich die Service-Welt auf meiner Firewall sperre, da geht nichts durch.
Das Problem ist erst seit geraumer Zeit aufgetreten, weiss nicht ab welcher Version wenn ich ehrlich bin.

**Comment by chmaod** [comment]
Leider nützt auch der Adapterneuinstall nix, auch Versionsrückspulung nicht wirklich.
Beim Start des Adapter motzt er auch, über alle möglichen Felder, welche nun Objekte sein sollten (typ) aber als String daher kommen (wie es eigentlich schon immer war).

**Comment by chmaod** [comment]
Serviceportal : Nicht erreichbar, nochmals geprüft.
Mir kommt es fast so vor, als wäre es zeitglich mit der Umstellung von Sommer/Winterzeit aufgetreten, aber würds jetzt nicht mit Blut untschreiben, aber kommt etwa hin, denn da hatte ich auch das erste mal im Log, dass das ISG Zickt, denn ich habe mir ein Skript geschrieben, wenn die Zeitabweichung vom ISG zu meinem IOB grösser 1h ist, ich eine Meldung erhalte (heartbeat)

**Comment by chmaod** [comment]
Zeit und Datum sind aber korrekt, auch gerade geprüft

**Comment by ben19840425** [comment]
Hallo, ich habe das gleiche Problem.

![image](https://user-images.githubusercontent.com/59473548/211169819-63b13095-3cb8-4b2f-828e-b6cc58c32fff.png)

**Comment by ben19840425** [comment]
Hallo,
Wird der Adapter nochmals weiter entwickelt?
Zu Schade, dass er mit den Fehlern aktuell nicht benutzbar ist.

Grüße

**Comment by unltdnetworx** [comment]
Im Reiter "URLs" der Instanz des Adapter sind verschiedene Zahlen getrennt durch ; aufgeführt. Das sind die Seiten, welche der Adapter vom ISG abholt bzw. versucht abzuholen.
Wenn man die Weboberfläche des ISG aufruft, enden die Seiten dort (URL) mit "/?=" gefolgt von Zahlen. Das sind diese erwähnten Zahlen. Also z.B. "/?=1,0"
Bitte mal die Zahlen überprüfen, indem ihr alles rauskopiert und einzeln wieder ergänzt, solange bis der Fehler wieder auftritt.
Es können auch nicht benötigte Seite weggelassen werden, was auch den Abruf beschleunigt. Ob Seiten bei euch überhaupt Ergebnisse liefern können, lässt sich manuell überprüfen. Einfach die Weboberfläche des ISG im Browser aufrufen und die einzelnen Zahlen aus der URLs-Seite des Adapters anhängen und mit Enter bestätigen. Wenn eine leere Seite im Browser auftaucht, kann diese Zahl im Adapter weggelassen werden.

Da es verschiedene Heizungen gibt müsste ich diesen Weg wählen, um verschiedene Typen abdecken zu können.

**Comment by unltdnetworx** [comment]
Das ist erst mal nur ein Workaround und wird hoffentlich zeitnah gefixt.

**Comment by ben19840425** [comment]
Hallo,
mit folgenden Einstellungen scheint es jetzt zu laufen (ISG Version: v12.2.3)

**command paths:**
0;4,0,0;4,0,1;4,1,0;4,1,1;4,2,0;4,2,4;4,3;4,3,1

**value paths:**
1,0;1,1;1,2;2,3

**status paths:**
1,0;1,1

**expert paths** habe ich komplett leer gelassen -> für mich keinen Mehrwert.


Was ist eigentlich der genaue Unterschied zwischen command/value/status path?

1,0 habe ich bei value und status paths

Grüße
Ben

**Comment by unltdnetworx** [comment]
Hallo Ben,
command paths sind Seiten, auf denen in den Weboberfläche etwas einstellbar ist, value paths beinhalten reine lesbare Infos wie z.B. Außentemperaturen o.Ä., Status waren glaub ich nicht einstellbare Werte der Heizung selbst. Die wurden in der Weboberfläche alle etwas anders definiert und mussten daher getrennt abgerufen werden.

**Comment by ben19840425** [comment]
> Hallo Ben, command paths sind Seiten, auf denen in den Weboberfläche etwas einstellbar ist, value paths beinhalten reine lesbare Infos wie z.B. Außentemperaturen o.Ä., Status waren glaub ich nicht einstellbare Werte der Heizung selbst. Die wurden in der Weboberfläche alle etwas anders definiert und mussten daher getrennt abgerufen werden.


Gibt es die Möglichkeit, die Werte zu ändern?
Aktuell wird der Wert "TRUE" geschrieben, wenn der Verdichter an ist. 
"NULL" wird geschrieben, wenn der Verdichter aus ist.

Leider kann die InfluxDB nicht mit NULL values umgehen. Kann man den Adapter dahingehend ändern, das "FALSE" zurückgegeben wird, wenn z.B. der Verdichter aus ist?
![image](https://github.com/unltdnetworx/ioBroker.stiebel-isg/assets/59473548/ad97172a-b70f-487e-a2d2-a3ccfb31da71)

**Comment by ben19840425** [comment]
@tipp88 funktioniert nicht. Sobald der Wert des Stiebel Datenpunkt (null) ist, bekomme ich im JavaScript log: getState "stiebel-isg.0.Info.STATUSANLAGE.BETRIEBSSTATUS.ABTAUEN" not found (3)

Mein Script sieht so aus:
`f (getState('stiebel-isg.0.Info.STATUSANLAGE.BETRIEBSSTATUS.ABTAUEN').val == true) {
  setStateDelayed('0_userdata.0.WPL20AC.wpl20ac_Abtauen', true, true, parseInt(((0) || '').toString(), 10), false);
} else {
  setStateDelayed('0_userdata.0.WPL20AC.wpl20ac_Abtauen', false, true, parseInt(((0) || '').toString(), 10), false);
} `

Hast du eine Idee?
Grüße

**Comment by tipp88** [comment]
https://blog.unltd-networx.de/iobroker-stiebel-eltron-tecalor-waermepumpe-per-modbus-anbinden

**Comment by ben19840425** [comment]
> https://blog.unltd-networx.de/iobroker-stiebel-eltron-tecalor-waermepumpe-per-modbus-anbinden

@tipp88 
habe ich jetzt mal umgestellt auf Modbus. Vielen Dank für den Link.
Weisst du wo ich die Aufnahmeleistung für den Inverter(Verdichter) im Register finde? Im Stiebel-Adapter wird es mit Kw angegeben. z.B. 1,5 Kw 

Danke.

**Comment by tipp88** [comment]
> > https://blog.unltd-networx.de/iobroker-stiebel-eltron-tecalor-waermepumpe-per-modbus-anbinden
> 
> @tipp88 habe ich jetzt mal umgestellt auf Modbus. Vielen Dank für den Link. Weisst du wo ich die Aufnahmeleistung für den Inverter(Verdichter) im Register finde? Im Stiebel-Adapter wird es mit Kw angegeben. z.B. 1,5 Kw
> 
> Danke.

Das wird über Modbus leider nicht geliefert. Verbrauchswerte lese ich weiterhin über den ISG Adapter aus, aber man kann dadurch die URLs in den Einstellungen drastisch reduzieren. Generell gilt alles über Modbus lesen und nur das allernötigste über den ISG Adapter.
